libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -0,0 +1,443 @@
+from libpermian.result import Result
+from libpermian.workflows.isolated import IsolatedWorkflow
+from libpermian.events.structures.base import BaseStructure
+from libpermian.plugins.compose import ComposeStructure
+from libpermian.exceptions import StructureConversionError
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@api.events.register_structure('InstallationSource')
+class InstallationSourceStructure(BaseStructure):
+    """ Structure with URLs and paths needed for compose installation
+
+    Example:
+    {"base_repo": "BaseOS",
+     "repos": {"BaseOS": {"x86_64": "http://example.com/compose/x86_64/BaseOS/os",
+                          "aarch64": "http://example.com/compose/aarch64/BaseOS/os"
+               "AppStream": {"x86_64": "http://example.com/compose/x86_64/AppStream/os",
+                             "aarch64": "http://example.com/compose/aarch64/AppStream/os"}},
+     "kernel_path": "images/pxeboot/vmlinuz",
+     "initrd_path": "images/pxeboot/initrd.img"}
+    """
+    def __init__(self, settings, base_repo, repos, kernel_path=None, initrd_path=None):
+        super().__init__(settings)
+        self.base_repo = base_repo
+        self.repos = repos
+        self.kernel_path = kernel_path
+        self.initrd_path = initrd_path
+
+        if self.base_repo not in self.repos.keys():
+            raise Exception('Invalid structure data, base_repo not in repos')
+
+    @classmethod
+    def from_compose(cls, compose):
+        """ Conversion method from ComposeStructure """
+        settings = compose.settings
+
+        if compose.product in ['RHEL']:
+            base_repo = 'BaseOS'
+        elif compose.product in ['Fedora']:
+            base_repo = 'Everything'
+        else:
+            raise StructureConversionError(ComposeStructure,
+                    InstallationSourceStructure,
+                    f'Product "{compose.product}" not supported')
+
+        repos = dict()
+        for variant in compose.composeinfo.metadata.info.get_variants():
+            repos[variant.id] = variant.paths.os_tree
+
+        return cls(settings, base_repo, repos, compose.composeinfo.kernel_path, compose.composeinfo.initrd_path)
+
+
+class ExecutionContainer():
+    """ Handles creation, removal of container image and running commands in it """
+    def __init__(self):
+        self.image_name = 'anaconda-webui'
+
+        img = subprocess.check_output(['podman', 'images', '-q', '-f', f'reference={self.image_name}']).decode()
+        if img == '':
+            containerfile = os.path.join(os.path.dirname(__file__), 'Containerfile')
+            subprocess.check_call(['podman', 'build', '-f', containerfile, '-t', self.image_name])
+
+    def exec(self, cmd, volume, env={}, cwd='/root/workdir'):
+        env_params = []
+        for key, val in env.items():
+            env_params += ['-e', f'{key}={val}']
+
+        podman_cmd = ['podman', 'run', '--rm', '-it',
+                      '-v', f'{volume}:/root/workdir:Z',
```

Comment:
Putting the mode to the argument is improvement, however it would be better if the default was read-only or overlay. If it's needed to obtain files created by the container, it's still possible to achieve that by using overlay like: `podman run -v `pwd`/lower:/root/foo:O,upperdir=`pwd`/upper,workdir=`pwd`/workdir --rm -it quay.io/centos/centos:stream8` In such case, the newly created files are in `upper` directory.